smslibrary/src/main/java/com/eis/smslibrary/SMSCore.java

```diff
@@ -0,0 +1,43 @@
+package com.eis.smslibrary;
+
+import android.app.PendingIntent;
+import android.telephony.SmsManager;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.ArrayList;
+
+/**
+ * Wrapper for the {@link android.telephony} library.
+ * This class is only used to interface with the core sms library of Android.
+ *
+ * @author Luca Crema
+ */
+class SMSCore {
```

Comment:
SMSCore isn't a static class?

Reply:
Why so? It has static methods

Reply:
Yes, I saw, so is it necessary for it to remain non-static?

Reply:
To become static the class needs to be _nested_, and this is not the case because it's a _top-level_ class: https://stackoverflow.com/questions/7486012/static-classes-in-java

Reply:
Oh, ok I didn't have that clear in my mind, thanks!